### PR TITLE
[REVIEW] Remove horizontal lines from comment blocks

### LINKED ISSUE #1332: [DOC] cuDF doxygen HTML output is messed up
**Location of incorrect documentation**
The documentation here: https://docs.rapids.ai/api/libcudf/stable/ is pretty useless. 

**Describe the problems or issues found in the documentation**
 - The first page that comes up is the `cudf` namespace reference, which is not a useful starting point. It should start with the Public API (functions) 
 - This page is a jumble of classes, functions, and mixed up documentation for them all. It's hard to say how this is happening.
 - There should be an overview page that explains what libcudf is, what the basic types are, and gives a couple of usage examples, here: https://docs.rapids.ai/api/libcudf/stable/ (currently blank)

**Suggested fix for documentation**
Check for doxygen warnings and errors.
Then trace the generated documentation to figure out how it is getting put together, and figure out which wrong/missing doxygen tags or settings are causing it.

### PR DESCRIPTION
**Pull request by harrism**
Currently the API docs for libcudf shows the main page for NVTX3 as its main page. Also, many classes have errors in their rendering due to horizontal rules in comments and other issues.

This PR removes `/**--------------------` from all doxygen blocks in libcudf. Part fix for #1332

The following were moved to one of @davidwendt 's PRs:
 - ~excludes NVTX from documentation rendering~ 
 - ~Adds a main_page.md where we can add useful introductory documentation about the library.~

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: cpp/doxygen/Doxyfile [hunk 1]
```diff
@@ -811,7 +811,7 @@ RECURSIVE              = YES
 # Note that relative paths are relative to the directory from which doxygen is
 # run.
 
-EXCLUDE                =
+EXCLUDE                =
```
Comment (harrism) [inline cpp/doxygen/Doxyfile:814]:
```suggestion
EXCLUDE                =
```

**Comment by davidwendt** [comment]
> Looks like you need a rebase Mark. Seems like you have a bunch of clang-format changes in your PR.

Are you sure? His last commit says "Fix clang-formatting" and I would expect CI would've failed on that push.

**Comment by codereport** [comment]
> > Looks like you need a rebase Mark. Seems like you have a bunch of clang-format changes in your PR.
> 
> Are you sure? His last commit says ["Fix clang-formatting"](https://github.com/rapidsai/cudf/pull/4809/commits/46780dcda86dc501edeebed6f99f46eccba3c611) and I would expect CI would've failed on that push.

You are right, I was looking at a stale version of the PR. Thanks!

**Pull request merged**